### PR TITLE
fix and update @doc warning and duplicate, fix dialyzer typespec

### PR DESCRIPTION
**Pull request by shahryarjb**
Hi dear @alisinabh I updated some lines of `jalaali.ex` and improve @doc and update its library and test it on elixir 1.12.0 and 1.12.2 and it has no error for @doc anymore. but it should be noted we have many errors for ( Calendar behaviour and test file ) which needs to be considered.

I am not good on Time module in elixir and erlang to fix this.

Thanks

### CODE REVIEW

File: mix.exs [hunk 1]
```diff
@@ -7,7 +7,7 @@ defmodule Jalaali.Mixfile do
     [
       app: :jalaali,
       version: @version,
-      elixir: "~> 1.3",
+      elixir: "~> 1.12.0",
```
Comment (alisinabh) [inline mix.exs:10]: Please revert this. 

We should use something like `"~> 1.6"` as if you use `"~> 1.12"` I think we will loose backwards compatibility with older versions of elixir. Generally only increase this if you use a functionality which is only existing in the newer version. But this needs to be done in another PR with and investigated on its own.

**Review by alisinabh — CHANGES_REQUESTED** [review]

File: lib/jalaali.ex [hunk 2]
```diff
@@ -11,92 +11,64 @@ defmodule Jalaali do
           2097, 2192, 2262, 2324, 2394, 2456, 3178]
 
   @doc """
```
[the remark this answers is not in the document]
Comment (alisinabh) [reply lib/jalaali.ex:13]: That makes sense. Thanks.

File: mix.exs (hunk 1 above)
Comment (shahryarjb) [reply mix.exs:10]: Unfortunately,  I have no older elixir version to test it, but as I saw nothing was deprecated I think it is compatible with elixir 1.6

I will try to update full version of this lib for elixir 1.12, it should be noted we have `22 doctests, 13 tests, 4 failures`

File: mix.exs [hunk 3]
```diff
@@ -7,7 +7,7 @@ defmodule Jalaali.Mixfile do
     [
       app: :jalaali,
       version: @version,
-      elixir: "~> 1.12.0",
+      elixir: "~> 1.6",
```
Comment (shahryarjb) [inline mix.exs:10]: can you test with 1.6 please
